src/System.Runtime.InteropServices.RuntimeInformation/src/ArchitectureManager.Unix.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ComponentModel;
+
+namespace System.Runtime.InteropServices
+{
+    internal partial struct ArchitectureManager
+    {
+        private static Architecture ProcessArchCore
+        {
+            get
+            {
+                if (IsARM)
+                {
+                    return Architecture.ARM32;
+                }
+                else
+                {
+                    return _is32BitProcess ? Architecture.BIT32 : Architecture.BIT64;
+                }
+            }
+        }
+
+        private static Architecture OSArchCore
+        {
+            get
+            {
+                string arch = null;
+                int result = Interop.Sys.UName(out arch);
+                if (result >= 0 && arch != null)
+                {
+                    if (arch.IndexOf("arm", StringComparison.OrdinalIgnoreCase) >= 0)
+                        return Architecture.ARM32;
```

Comment:
Right now, we are not building for arm64, but I get that we will have to support it soon. One option is to make Architecture similar to OSPlatform, and have a struct OSArchitecture, with x86, x64, ARM32, ARM64, and depend on the build time constants on Unix, and on the SYSTEM_INFO struct on windows.